### ISSUE TITLE
Occasional error when xonsh start - JSONDecodeError

### ISSUE THREAD

**Issue by mitnk** [xonsh/xonsh#1238]
```
hugo@mbp: ~$ Exception in thread Thread-2:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/threading.py", line 914,r
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/xonsh/ptk/n
    lj = lazyjson.LazyJSON(f, reopen=False)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/xonsh/lazy_
    self._load_index()
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/xonsh/lazyx
    locs = json.loads(locs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/__init__.py", line s
    return _default_decoder.decode(s)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/decoder.py", line 3e
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/decoder.py", line 3e
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

No investigations yet, just put some tracebacks here.

**Comment by JohnLunzer** [comment]
Not in Xonsh but I have had this error about 1000 times in another Python app which is trying to read a json file as it's being written to disk. This might be fixed by catching the JSONDecodeError (or ValueError for python 3.4) and retrying the load, but might be a chance it could fail again.

**Comment by laerus** [comment]
i also sometimes get a json related error when starting xonsh, dunno if this relates to this issue
```
Exception ignored in: <bound method LazyJSON.__del__ of <xonsh.lazyjson.LazyJSON object at 0x7f4603ad5400>>
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/xonsh-0.3.4-py3.5.egg/xonsh/lazyjson.py", line 213, in __del__
    self.close()
  File "/usr/local/lib/python3.5/dist-packages/xonsh-0.3.4-py3.5.egg/xonsh/lazyjson.py", line 218, in close
    self._f.close()
OSError: [Errno 9] Bad file descriptor
```

**Comment by scopatz** [comment]
Thanks for reporting these issues @mitnk and @laerus. I have never seen these.  Do you have any steps to reproduce?

**Comment by laerus** [comment]
@scopatz i can't find a way to reproduce this, it seems a little random, if i find out more i'll open a new issue

**Comment by scopatz** [comment]
Ok sounds like a plan!

**Comment by mitnk** [comment]
It's vary rare and usually happened after install a fresh commit of xonsh copy (but not every time). It just happened on my side again:

```
Last login: Thu Jun 23 11:05:31 on ttys011
hugo@mbp: ~$ Exception in thread Thread-3:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/xonsh/ptk/history.py", line 59, in run
    lj = lazyjson.LazyJSON(f, reopen=False)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 511, in __init__
    self._load_index()
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 539, in _load_index
    locs = json.loads(locs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/__init__.py", line 319, in loads
    return _default_decoder.decode(s)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

hugo@mbp: ~$
```

And, it didn't crash the xonsh session.

**Comment by JohnLunzer** [comment]
@scopatz, I realize it is unorthodox but I recommend handling this exception in the master branch which would allow more information to be reported (e.g. which file is being affected by this). Because it is rare you will need many users to get a hit. In the exception handling you can put a message to report the information to this issue thread. Again, I realize it is not common practice but

**Comment by scopatz** [comment]
> @scopatz, I realize it is unorthodox but I recommend handling this exception in the master branch

This exception used to be handled and then someone removed it for reasons I don't fully understand...

**Comment by scopatz** [comment]
A fix is now in #1328
